src/Symfony/Bridge/PhpUnit/ExpectUserDeprecationMessageTrait.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+use PHPUnit\Runner\Version;
+
+if (version_compare(Version::id(), '11.0.0', '<')) {
+    trait ExpectUserDeprecationMessageTrait
+    {
+        use ExpectDeprecationTrait;
+
+        final protected function expectUserDeprecationMessage(string $expectedUserDeprecationMessage): void
+        {
+            $this->expectDeprecation($expectedUserDeprecationMessage);
```

Comment:
Be careful. The PHPUnit 11 API performs an exact match of the deprecation message. `expectDeprecation` uses `assertStringMatchesFormat`. So this is not a good polyfill (tests that rely on format placeholders would break when using PHPUnit 11 which performs exact match)

Reply:
Is there a way to escape those placeholders?

Reply:
you can escape by doubling `%` (like in `sprintf` patterns)

Reply:
So, I replace all `%` with `%%` and I should be good?

Reply:
it should, yes

Reply:
#58317